### PR TITLE
add: orWhere & andWhere methods

### PR DESCRIPTION
**Pull request by stillalivx**
I added the orWhere and andWhere methods. I read your comment about a cool API to add these clauses like "automatically", but I thing that this is a good temportal solution. The query or & and are very usefull and denodb has to include them (also to close issue #197)

Ah, in addition, I left a little issue (#280) about a weird behavior, I hope you can check it out!

### CODE REVIEW

File: lib/query-builder.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export type QueryType =
   | "max"
   | "avg"
   | "sum";
+export type WhereMethod = "none" | "and" | "or";
```
Comment (eveningkid) [inline lib/query-builder.ts:21]: I don't think that `none` should be represented as a string here but `undefined` instead (and come last). This should also be changed in other files (I've seen it in `model.ts`, not sure about others)

File: lib/query-builder.ts [hunk 2]
```diff
@@ -178,15 +180,17 @@ export class QueryBuilder {
...
     operator: Operator,
     value: FieldValue,
+    method: WhereMethod,
   ) {
     if (!this._query.wheres) {
       this._query.wheres = [];
     }
 
-    const whereClause = {
+    const whereClause: WhereClause = {
```
Comment (eveningkid) [inline lib/query-builder.ts:189]: Just wondering: would it trigger an error if there was no `: WhereClause` here?

File: lib/translators/sql-translator.ts [hunk 3]
```diff
@@ -78,11 +78,31 @@ export class SQLTranslator implements Translator {
 
     if (query.wheres) {
       query.wheres.forEach((where) => {
-        queryBuilder = queryBuilder.where(
-          where.field,
-          where.operator,
-          where.value,
-        );
+        switch (where.method) {
+          case "none":
```
Comment (eveningkid) [inline lib/translators/sql-translator.ts:82]: So this should become the default case instead

File: lib/query-builder.ts (hunk 1 above)
Comment (eveningkid) [reply lib/query-builder.ts:21]: Oh and btw, could you add an extra line between `WhereMethod` and the previous `QueryType`?

**Review by eveningkid — CHANGES_REQUESTED** [review]
Hey,

It's been a while and to be honest with you, I opened this PR a few days ago and completely forgot to send you my comments!

So here there are. I think there's not so much to change, it's more about refactoring your code and we should be good.

This is a really good feature, thank you for your work!

Keep in touch

File: lib/model.ts [hunk 4]
```diff
@@ -629,6 +630,121 @@ export class Model {
...
+          this.formatFieldToDatabase(field) as string,
+          "=",
+          value,
+          "and",
+        );
+      }
+    }
+
+    return this;
+  }
```
[the remark this answers is not in the document]
Comment (stillalivx) [reply lib/model.ts:694]: Hi! I'm having problems traying to implement this part. In the methods where, andWhere & orWhere there are a conditions to verify the arguments. That is a problem 'cause is necessary in every method. I tried to pass all the arguments with a spread operator to `_where`:
```js
private static _where(...) {
  if (...) {
    ...
    this._currentQuery.where(...);
  } else {
    ...
    this._currentQuery.where(...);
  }
}

static andWhere(...) {
  const args = Array.from(arguments) // If I pass arguments directly, it throws an error (arguments is not an iterable)
  return this._where("and", ...args); // Throws error "A spread argument must either have a tuple type or be passed to a rest parameter"
}
```
But that throws a syntax error. I can't see how to proceed.

**Pull request closed without merging**